opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py

```diff
@@ -32,42 +32,43 @@ def extract_attributes_from_object(
     return extracted
 
 
-def http_status_to_canonical_code(
+def http_status_to_status_code(
     status: int, allow_redirect: bool = True
-) -> StatusCanonicalCode:
+) -> StatusCode:
     """Converts an HTTP status code to an OpenTelemetry canonical status code
 
     Args:
         status (int): HTTP status code
     """
     # pylint:disable=too-many-branches,too-many-return-statements
+    # See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#status
```

Comment:
I feel like this section could be refactored to a much simpler tree of conditions.

Reply:
Perhaps. I left it in there just to show the possible httpstatuses.